hotshot-query-service/src/fetching/provider/query_service.rs

```diff
@@ -56,68 +55,56 @@ impl<Ver: StaticVersionType> QueryServiceProvider<Ver> {
 impl<Ver: StaticVersionType> QueryServiceProvider<Ver> {
     async fn fetch_legacy_payload<Types: NodeType>(
         &self,
+        payload_bytes: Vec<u8>,
+        common_bytes: Vec<u8>,
         req: PayloadRequest,
     ) -> Option<Payload<Types>> {
         let PayloadRequest(VidCommitment::V0(advz_commit)) = req else {
             return None;
         };
 
-        let res = try_join!(
-            self.client
-                .get::<ADVZPayloadQueryData<Types>>(&format!(
-                    "availability/payload/hash/{}",
-                    advz_commit
-                ))
-                .send(),
-            self.client
-                .get::<ADVZCommonQueryData<Types>>(&format!(
-                    "availability/vid/common/payload-hash/{}",
-                    advz_commit
-                ))
-                .send()
-        );
-        match res {
-            Ok((payload, common)) => {
-                let num_storage_nodes = ADVZScheme::get_num_storage_nodes(common.common()) as usize;
-                let bytes = payload.data.encode();
-                let commit = match advz_scheme(num_storage_nodes).commit_only(bytes) {
-                    Ok(commit) => commit,
-                    Err(err) => {
-                        tracing::error!(%err, "unable to compute VID commitment");
-                        return None;
-                    },
-                };
-                if commit != advz_commit {
-                    tracing::error!(?req, ?commit, "received inconsistent payload");
-                    return None;
-                }
+        let payload =
```

Comment:
Rename this function to `decode_legacy_payload` or similar?

Reply:
https://github.com/EspressoSystems/espresso-network/pull/3175/commits/cc91b39dcb0248118394e8973fb26f5fa25429d0